R/R/install.R

```diff
@@ -0,0 +1,150 @@
+#' **metaflow** and it's Python dependencies.
+#'
+#' This package uses the [reticulate][reticulate::reticulate] package
+#' to make an interface with the [Metaflow](https://metaflow.org/)
+#' Python package.
+#'
+#' @param method `character`, indicates to use `"conda"` or `"virtualenv"`.
+#' @param version `character`, version of Metaflow to install. The default version
+#' is the latest available on PyPi.
+#' @param ... other arguments sent to [reticulate::conda_install()] or
+#'    [reticulate::virtualenv_install()]
+#'
+#' @seealso
+#' [reticulate: Using reticulate in an R Package](https://rstudio.github.io/reticulate/articles/package.html),
+#' [reticulate: Installing Python Packages](https://rstudio.github.io/reticulate/articles/python_packages.html)
+#' @examples
+#' \dontrun{
+#' # not run because it requires Python
+#' install_metaflow()
+#' }
+#' @export
+install_metaflow <- function(method = c("conda", "virtualenv"),
+                             version = NULL,
+                             ...) {
+  envname <- pkg.env$envname
+
+  env_set <- check_environment(envname)
+  if (method == "conda" && env_set[["virtualenv"]]){
+    message("Installing Metaflow python backend using conda but it has been installed in virtualenv.")
+    message("We do not allow python backend in both conda and virtualenv.")
+    message("1. To reinstall python backend in virtualenv: install_metaflow(method='virtualenv')")
+    message("2. To remove python backend: remove_metaflow()")
+    stop("Installing python backend in both conda and virtualenv is not allowed.")
+  }
+
+  if (method == "virtualenv" && env_set[["conda"]]){
+    message("Installing Metaflow python backend using virtualenv but it has been installed in conda.")
+    message("We do not allow python backend in both conda and virtualenv.")
+    message("1. To reinstall python backend in conda: install_metaflow(method='conda')")
+    message("2. To remove python backend: remove_metaflow()")
+    stop("Installing python backend in both conda and virtualenv is not allowed.")
+  }
+
+  # validate stage, method arguments
+  method <- match.arg(method)
+
+  # conda and pip use different syntax for indicating versions
+  if (identical(method, "conda")) {
+    version_sep <- "="
+  } else {
+    version_sep <- "=="
+  }
+
+  if (is.null(version)) {
+    metaflow_pkg_version <- "metaflow"
+  } else {
+    metaflow_pkg_version <- paste("metaflow", version, sep = version_sep)
+  }
+
+  packages <- c(metaflow_pkg_version, "numpy", "pandas")
+
+  # create environment if not present
+  if (method == "conda"){
+      conda <- tryCatch(reticulate::conda_binary(), 
+                        error = function(e) NULL)
+      have_conda <- !is.null(conda)
+      if (!have_conda) {
+        message("No conda was found in the system.")
+        message("Would you like to download and install Miniconda?")
```

Comment:
Delete L:72 and L:73

Reply:
We can delete L:72, but L:73 should be there? L:73 and L:74 kind of should go together to provide info about Miniconda which we are asking to user to install